spec/features/user_can_cancel_correction_spec.rb

```diff
@@ -0,0 +1,28 @@
+require 'rails_helper'
+
+RSpec.feature 'Cancelling a correction submission' do
+  around(:each) do |example|
+    ClimateControl.modify CORRECTION_RETURNS_ENABLED: 'yes' do
+      example.run
+    end
+  end
+
+  context 'for a task with a correction in progress' do
+    scenario 'it can be cancelled' do
+      visit '/'
+
+      mock_user_endpoint!
+      mock_sso_with(email: 'email@example.com')
+      mock_incomplete_tasks_endpoint!
+      click_link 'sign-in'
+
+      mock_task_with_framework_endpoint!(task_id: 'b847e0f7-027e-4b95-afa2-3490b8d05a1d')
```

Comment:
the various mock calls in this spec were difficult to follow. it helps that they appear before the action that requires them. It might also be useful if this magic task id is given a name, e.g. `mock_corrrecting_task_id`.